src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleRequestSender.cs

```diff
@@ -770,5 +778,33 @@ private void HandleCustomHostLaunch(ITestHostLauncher customHostLauncher, Messag
                 this.communicationManager.SendMessage(MessageType.CustomTestHostLaunchCallback, ackPayload, this.protocolVersion);
             }
         }
+
+        private void AttachDebuggerToProcess(ITestHostLauncher customHostLauncher, Message message)
+        {
+            if (customHostLauncher == null)
+            {
+                throw new ArgumentNullException("VsTestConsoleRequestSender.AttachDebuggerToProcess: test host launcher cannot be null.");
+            }
+            if (!(customHostLauncher is IDebugTestHostLauncher launcher))
+            {
+                throw new NotSupportedException("VsTestConsoleRequestSender.AttachDebuggerToProcess: test host launcher is not ITestHostLauncher2.");
```

Comment:
As mentioned before, both exception messages may need to be localized if they are user facing and consider `Debug.Fail` if it is not possible to arrive here in practice.

Reply:
Removed both exceptions to stay in sync with the way HandleCustomHostLaunch method is implemented.